modules/howtos/examples/kv_operations.py

```diff
@@ -1,240 +1,145 @@
-xref:6.5@server:learn:buckets-memory-and-storage/expiration.adoc#expiration-bucket-versus-item[Expiration Overview page].
-
-[source,python]
-                       ----
-"""
-# tag::expiration[]
-document = {"foo": "bar", "bar": "foo"}
-result = collection.upsert("document-key", document, cas=12345, expiry=timedelta(minutes=1),
-                           durability=ClientDurability(ReplicateTo.ONE, PersistTo.ONE))
-# end::expiration[]
-"""
-----
-
-Here, we have add _Durability_ options, namely `PersistTo` and `ReplicateTo`.
-    In Couchbase Server releases before 6.5, Durability was set with these two options -- see the xref:https://docs.couchbase.com/python-sdk/2.5.5/durability.html[6.0 Durability documentation] -- covering  how many replicas the operation must be propagated to and how many persisted copies of the modified record must exist.
-    Couchbase Data Platform 6.5 refines these two options, with xref:synchronous-replication.adoc[Synchronous Replication] -- although they remain essentially the same in use -- as well as adding the option of xref:transactions.adoc[atomic document transactions].
 
-The following example demonstrates using the newer durability features available in Couchbase server 6.5 onwards.
-
-[source,python]
-----
-"""
-# tag::upsert_syncrep[]
-from couchbase.durability import Durability
+from couchbase.durability import Durability, ServerDurability, \
+    ClientDurability, ReplicateTo, PersistTo
+from couchbase.exceptions import CouchbaseException
+from couchbase.cluster import Cluster
+from couchbase.auth import PasswordAuthenticator
+from couchbase.collection import InsertOptions, ReplaceOptions, \
+    UpsertOptions, GetOptions, RemoveOptions, IncrementOptions, \
+    DecrementOptions
+from couchbase.collection import DeltaValue, SignedInt64
+
+cluster = Cluster(
+    "couchbase://localhost",
+    authenticator=PasswordAuthenticator(
+        "username",
+        "password"))
+bucket = cluster.bucket("default")
+collection = bucket.default_collection()
+
+# tag::insert[]
+# Insert document
+document = {"foo": "bar", "bar": "foo"}
+result = collection.insert("document-key", document)
+# end::insert[]
+print("Result: {}; CAS: {}".format(result, result.cas))
 
+# tag::insert_w_opts[]
+# Insert document with options
+document = {"foo": "bar", "bar": "foo"}
+opts = InsertOptions(timeout=timedelta(seconds=5))
+result = collection.insert("document-key-opts",
+                           document,
+                           opts,
+                           expiry=timedelta(seconds=30))
+# end::insert_w_opts[]
+
+try:
+    # tag::replace_cas[]
+    # Replace document with CAS
+    document = {"foo": "bar", "bar": "foo"}
+    result = collection.replace(
+        "document-key",
+        document,
+        cas=12345,
+        timeout=timedelta(
+            minutes=1))
+# end::replace_cas[]
+except CouchbaseException as ex:
+    # we expect an exception here as the CAS value is chosen
+    # for example purposes
```

Comment:
I'd be tempted to do a retrieve and use a variable for cas, to give a full round-trip example for how to use it (mostly as it confused me personally for a bit... the extra retrieve doesn't need to go in the code sample, but can be referred to by clicking through.)

(I'm happy to make that change if you think it seems reasonable, let me know.)

Reply:
I followed the Go SDK docs, there is an example showing CAS (just a number -- 12345) and then a more "meaningful" example (get followed by replace passing in CAS).  I am up for whatever seems best.  Just let me know.